index.js

```diff
@@ -1,165 +1,116 @@
+import "./css/styles.css";
 
-const albumBucketName;
-const cloudfrontBaseUrl;
-const identityPoolId;
-const awsRegion;
+const albumBucketName = process.env.BUCKET_NAME;
+const cloudfrontBaseUrl = process.env.CLOUD_FRONT_BUCKET_URL;
+const identityPoolId = process.env.IDENTITY_POOL_ID;
+const awsRegion = process.env.AWS_REGION;
 
+// DOM Elements
+const photosContainerEl = document.querySelector("#photos");
 
 // Initialize the Amazon Cognito credentials provider
 AWS.config.region = awsRegion; // Region
 AWS.config.credentials = new AWS.CognitoIdentityCredentials({
-    IdentityPoolId: identityPoolId,
+  IdentityPoolId: identityPoolId,
 });
 
 // Create a new service object
 const s3 = new AWS.S3({
-    apiVersion: '2006-03-01',
-    params: {Bucket: albumBucketName}
+  apiVersion: "2006-03-01",
+  params: { Bucket: albumBucketName },
 });
 
-// A utility function to create HTML.
-function getHtml(template) {
-    return template.join('\n');
-}
-
+// Delegated events
+photosContainerEl.addEventListener("click", (event) => {
+  if (event.target.className === "backtoalbums") {
+    listAlbums();
+  }
+  if (event.target.className === "pagebutton") {
+    viewAlbum(event.target.dataset.albumname, event.target.dataset.pagetag, event.target.dataset.prevtag);
+  }
+  if (event.target.className === "album") {
+    viewAlbum(event.target.dataset.albumname);
+  }
+});
 
 // Show the photos that exist in an album.
-function viewAlbum(albumName, startAfter = albumName+"/jpg/", prevTag = "") {
-    var albumPhotosKey = albumName + '/';
-    s3.listObjectsV2({Prefix: albumPhotosKey+"jpg/", StartAfter: startAfter, MaxKeys: 30}, function(err, data) {
-        if (err) {
-            return alert('There was an error viewing your album: ' + err.message);
-        }
-        // 'this' references the AWS.Response instance that represents the response
-        var href = this.request.httpRequest.endpoint.href;
-        var photoKey = "";
-        var i = 0;
```

Comment:
what was the intention of this?

Reply:
I was thinking about you and Frankie watching Sesame Street yesterday and wanted to try my hand at counting.